compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -478,10 +483,71 @@ static bool canUseInOperandAsInitOperand(OpOperand *inOperand,
   return true;
 }
 
+// Intentionally simplistic cost model for linalg ops: returns the number of
+// loop iterations times a measure of the cost of the block, which is
+// the number of ops in the block plus the number of arguments; adding the
+// number of arguments is done to account for the cost of reading elements from
+// memory. We are not nearly at the level of details where we would want to
+// account for relative cost of different ops, and we couldn't do that here
+// anyway.
+//
+// TODO: Should use APInt instead of int64 to avoid any risk of overflow; though
+// that is unlikely to happen as long as kDynSizeEstimate is small.
+static int64_t evalCost(linalg::LinalgOp op) {
+  // For dynamic dims, we have to arbitrarily pick a value. Fortunately, at the
+  // coarse level of detail that we are working at, the exact value should not
+  // matter --- the very basic fusion decisions made here should be the same
+  // for all but the tiniest sizes (it's plausible that values <=4 would affect
+  // the decision).
+  const int64_t kDynSizeEstimate = 16;
+  int64_t itersEstimate = 1;
+  for (int64_t r : op.getStaticLoopRanges()) {
+    itersEstimate *= (r == ShapedType::kDynamic) ? kDynSizeEstimate : r;
+  }
+  Block *block = op.getBlock();
+  int64_t blockCost = block->getNumArguments();
+  for (const auto &unused : block->getOperations()) {
+    (void)unused;
+    ++blockCost;
+  }
+  return itersEstimate * blockCost;
+}
+
+// Returns the cost of the fused op, or failure if we were not able to evaluate
+// it.
+static FailureOr<int64_t> evalFusedCost(MLIRContext *context,
+                                        linalg::LinalgOp producerOp,
+                                        linalg::LinalgOp consumerOp) {
+  OpBuilder b(context);
+  for (OpResult producerResult : producerOp->getOpResults()) {
+    const auto &uses = OpOperand::getUseList(producerResult)->getUses();
+    for (auto &use : uses) {
+      if (use.getOwner() == consumerOp) {
+        auto info = linalg::fuseProducerOfTensor(b, producerResult, use);
```

Comment:
This does not belong here. This is related to elementwise operation fusion that has already been done before this stage. Here we are trying to fuse using tile + fuse.